x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/custom_grid_body.tsx

```diff
@@ -0,0 +1,116 @@
+import type { DataTableRecord } from '@kbn/discover-utils/types';
+import type { EuiTheme } from '@kbn/react-kibana-context-styled';
+import type { TimelineItem } from '@kbn/timelines-plugin/common';
+import type { FC } from 'react';
+import React, { useMemo } from 'react';
+import styled from 'styled-components';
+import type { RowRenderer } from '../../../../../../common/types';
+import { useStatefulRowRenderer } from '../../body/events/stateful_row_renderer/use_stateful_row_renderer';
+
+export type RenderCustomBodyProps = EuiDataGridCustomBodyProps & {
+  rows: Array<DataTableRecord & TimelineItem> | undefined;
+  enabledRowRenderers: RowRenderer[];
+};
+
+export const RenderCustomBody: FC<RenderCustomBodyProps> = (props) => {
+  const { Cell, visibleColumns, visibleRowData, rows, enabledRowRenderers } = props;
+
+  const visibleRows = useMemo(
+    () => (rows ?? []).slice(visibleRowData.startRow, visibleRowData.endRow),
+    [rows, visibleRowData]
+  );
+
+  return (
+    <>
+      {visibleRows.map((row, rowIndex) => (
+        <RenderCustomSingleRow
+          rowData={row}
+          rowIndex={rowIndex}
+          key={rowIndex}
+          visibleColumns={visibleColumns}
+          Cell={Cell}
+          enabledRowRenderers={enabledRowRenderers}
+        />
+      ))}
+    </>
+  );
+};
+
+const CustomGridRow = styled.div.attrs<{
+  className?: string;
+}>((props) => ({
+  className: `euiDataGridRow ${props.className ?? ''}`,
+  role: 'row',
+}))`
+  width: fit-content;
+  border-bottom: 1px solid ${(props) => (props.theme as EuiTheme).eui.euiBorderThin};
+`;
+
+/* below styles as per : https://eui.elastic.co/#/tabular-content/data-grid-advanced#custom-body-renderer */
+const CustomGridRowCellWrapper = styled.div.attrs({ className: 'rowCellWrapper', role: 'row' })`
+  display: flex;
+`;
+
+type RenderCustomSingleRowProps = {
+  rowData: DataTableRecord & TimelineItem;
+  rowIndex: number;
+} & Pick<RenderCustomBodyProps, 'visibleColumns' | 'Cell' | 'enabledRowRenderers'>;
+
+const RenderCustomSingleRow = (props: RenderCustomSingleRowProps) => {
+  const { rowIndex, rowData, enabledRowRenderers, visibleColumns, Cell } = props;
+
+  const { canShowRowRenderer } = useStatefulRowRenderer({
+    data: rowData.ecs,
+    rowRenderers: enabledRowRenderers,
+  });
+
+  // removes the border between the actual row and Additional row
+  // which renders the AdditionalRow
+  const cellCustomStyle = useMemo(
+    () =>
+      canShowRowRenderer
+        ? {
+            borderBottom: 'none',
+          }
+        : {},
+    [canShowRowRenderer]
+  );
+
+  return (
+    <CustomGridRow
+      className={`${rowIndex % 2 === 0 ? 'euiDataGridRow--striped' : ''}`}
+      key={rowIndex}
+    >
+      <CustomGridRowCellWrapper>
+        {visibleColumns.map((column, colIndex) => {
+          // Skip the row details cell - we'll render it manually outside of the flex wrapper
+          if (column.id !== 'additional-row-details') {
```

Comment:
can we give the id a more unique name like `timeline-expanded-row-details` or something to that effect. It shouldn't really affect anything since we aren't targeting any styles to the id, but in case eui uses something generic like this in the future..

Reply:
Makes sense. I named it `timeline-event-detail-row`. Because expanded means something different ( when event details flyout is opened ). See if you are okay with this or you have abetter name.

Review changes here : be6bc648b8a

Reply:
Lol event detail could be potentially confusing, but as long as timeline is somewhere in there to add some specificity 👍🏾 . Thanks for making the change!